assets/src/stories-editor/components/editor-carousel/indicator.js

```diff
@@ -29,27 +36,100 @@ const Indicator = ( { pages, currentPage, onClick } ) => {
+
 	return (
 		<ul className="amp-story-editor-carousel-item-list">
 			{ pages.map( ( page, index ) => {
 				const className = page.clientId === currentPage ? 'amp-story-editor-carousel-item amp-story-editor-carousel-item--active' : 'amp-story-editor-carousel-item';
+				const blockElementId = `amp-story-editor-carousel-item-${ page.clientId }`;
+				const transferData = {
+					type: 'indicator',
+					srcIndex: getBlockIndex( page.clientId ),
+					srcClientId: page.clientId,
+				};
+
+				const getInsertIndex = ( position ) => {
+					const dstIndex = getBlockIndex( page.clientId );
+					if ( page.clientId !== undefined ) {
+						return position.x === 'left' ? dstIndex : dstIndex + 1;
+					}
+
+					return undefined;
+				};
+
+				const onDrop = ( event, position ) => {
+					const { srcClientId, srcIndex, type } = parseDropEvent( event );
+
+					const isIndicatorDropType = 'indicator' === type;
+					const isSameBlock = srcClientId === page.clientId;
+
+					if ( ! isIndicatorDropType || isSameBlock ) {
+						return;
+					}
+
+					const positionIndex = getInsertIndex( position );
+					const insertIndex = srcIndex < index ? positionIndex - 1 : positionIndex;
+					moveBlockToPosition( srcClientId, '', '', insertIndex );
+				};
+
+				const isPageDragged = page.clientId === draggedPage;
+				const isCurrentPage = page.clientId !== currentPage;
+				const indicatorButton = (
+					<Button
+						onClick={ ( e ) => {
+							e.preventDefault();
+							onClick( page.clientId );
+						} }
+					>
+						<span className="indicator-page-number">{ index + 1 }</span>
+						<span className="screen-reader-text">
+							{ label( index + 1 ) }
+						</span>
+					</Button>
+				);
 
 				return (
-					<li key={ page.clientId } className={ className }>
-						<Tooltip text={ toolTip( index + 1 ) }>
-							<Button
-								onClick={ ( e ) => {
-									e.preventDefault();
-									onClick( page.clientId );
-								} }
-								disabled={ page.clientId === currentPage }
-							>
-								<span className="screen-reader-text">
-									{ label( index + 1 ) }
-								</span>
-							</Button>
-						</Tooltip>
-					</li>
+					<Draggable
+						key={ page.clientId }
+						elementId={ blockElementId }
+						transferData={ transferData }
+						onDragStart={ () => {
+							setDraggedPage( page.clientId );
+						} }
+						onDragEnd={ () => {
+							setDraggedPage( null );
+						} }
+					>
+						{
+							( { onDraggableStart, onDraggableEnd } ) => (
+								<>
+									<li
+										className={ className }
+									>
+										<div
+											onDragStart={ onDraggableStart }
+											onDragEnd={ onDraggableEnd }
+											draggable
+											className="amp-story-editor-carousel-item-wrapper"
+											id={ blockElementId }
+										>
+											{ isCurrentPage ?
+												indicatorButton :
+												<Tooltip text={ toolTip( index + 1 ) }>{ indicatorButton }</Tooltip>
```

Comment:
The tooltip now only seems to be shown when focusing/hovering over the the current page, not the other ones.